pkg/apis/serving/fieldmask.go

```diff
@@ -609,17 +609,14 @@ func SecurityContextMask(ctx context.Context, in *corev1.SecurityContext) *corev
 	out := new(corev1.SecurityContext)
 
 	// Allowed fields
-	out.RunAsUser = in.RunAsUser
-	if in.RunAsNonRoot != nil && *in.RunAsNonRoot {
-		out.RunAsNonRoot = in.RunAsNonRoot
-	}
```

Comment:
IIRC we had some reason to only allow `true` through here. You sure that's not necessary?

Reply:
Left a comment below regarding this

```
	// RunAsNonRoot when unset behaves the same way as false
 	// We do want the ability for folks to set this value to true
```

Reply:
I think we were worried that people may have defaulted runAsNonRoot to true, in which case allowing it to be false would be a security regression. I _think_ the PSP doesn't allow you to default runAsNonRoot though (it just has a MustRunAsNonRoot setting instead) so this might be fine.

Reply:
.. in fact it's a straight boolean and not a pointer, so it can't really be defaulted any way other than false